init: reservation initial setting

zerone1202 opened the pull request:
## #️⃣ Issue Number<!--- ex) #이슈번호, #이슈번호 -->

- Closes #19 

## 📝 요약(Summary)<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- Reservation 도메인의 초기 세팅 작업
- Reservation 엔티티 생성 및 필드 정의
- hostUser / guestUser 연관관계 설정
- ReservationRepository 생성

## 🛠️ PR 유형
어떤 변경 사항이 있나요?
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## ✅ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성

zerone1202 committed:
init(reservation): Add Reservation entity

- Reservation 엔티티 생성
- reservationStartAt, reservationEndAt 필드 추가
- price, deposit, status 필드 생성
- hostUser, guestUser 연관관계(N:1) 설정
- post 추후 연관관계 설정 후 주석 해제 예정
- BaseEntity 상속 구조 적용

zerone1202 committed:
init(reservation): Add ReservationStatus enum

- 예약 상태 관리를 위한 ReservationStatus Enum 생성
- REQUESTED, APPROVED, RENTED, RETURNING, COMPLETED, CANCELLED, REJECTED 상태 정의

zerone1202 committed:
init(reservation): Add ReservationRepository

- ReservationRepository 생성
- 기본 CRUD 지원을 위해 JpaRepository 상속

zerone1202 committed:
Merge branch 'dev' of https://github.com/sparta-finale/golfrental into init/19-reservation
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

AllSungho commented on src/main/java/com/golfRental/domain/reservation/entity/Reservation.java:
```diff
@@ -0,0 +1,63 @@
+    private User guestUser;
+
+    @Builder
+    public Reservation(LocalDateTime reservationStartAt, LocalDateTime reservationEndAt, Integer price, Integer deposit, ReservationStatus status, User hostUser, User guestUser) {
```
Builder 사용을 위해 public이 아닌 prviate으로 접근 지정자 수정하는 것이 좋아보입니다!

zerone1202 replied:
바로 반영하겠습니다~

zerone1202 commented on src/main/java/com/golfRental/domain/reservation/entity/Reservation.java:
```diff
@@ -0,0 +1,63 @@
+    @Column(nullable = false)
+    private LocalDateTime reservationEndAt;
+
+    @Column(nullable = false)
+    private Integer price;
+
+    @Column(nullable = false)
+    private Integer deposit;
```
현재 금액은 정수 기반으로만 처리되고 소수점 계산이 없어 BigDecimal이 꼭 필요하지 않은 상황입니다.
변경 시 영향 범위가 커지기에, 초기 세팅 단계에서는 Integer 타입을 유지하겠습니다.
추후 금융 계산 로직이 추가되면 BigDecimal 도입 검토하겠습니다!

zerone1202 commented on src/main/java/com/golfRental/domain/reservation/entity/Reservation.java:
```diff
@@ -0,0 +1,63 @@
+        this.status = status;
+        this.hostUser = hostUser;
+        this.guestUser = guestUser;
+    }
```
엔티티 생성자에서 유효성 검증을 수행할 경우 도메인 책임이 커지고,  
예약 생성/수정/상태 전환 등 다양한 흐름에서 유연성이 떨어질 수 있다고 생각합니다!
검증 로직은 ReservationService에서 처리하는 방식으로 구현하겠습니다.

zerone1202 committed:
refactor(reservation): Change builder constructor to private

- @Builder 사용 시 외부에서 생성자를 호출하지 않도록 접근 제어자를 private으로 변경
- Reservation 엔티티의 생성 책임을 Builder로 명확히 분리하기 위함

AllSungho reviewed (approved):
커멘트 확인했고 수정사항 반영된 것까지 확인했습니다 수고하셨습니다!!!!!
